hydromt_wflow/wflow.py

```diff
@@ -90,19 +76,32 @@ class WflowModel(Model):
 
     def __init__(
         self,
-        root=None,
-        mode="w",
-        config_fn=None,
-        data_libs=None,
-        deltares_data=False,
+        root: Optional[str] = None,
+        mode: Optional[str] = "w",
+        config_fn: Optional[str] = None,
+        data_libs: Union[List, str] = None,
         logger=logger,
+        **artifact_keys,
     ):
+        for lib, version in artifact_keys.items():
+            warnings.warn(
+                "Adding a predefined data catalog as key-word argument is deprecated, "
+                f"add the catalog as '{lib}={version}'"
+                " to the data_libs list instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            if not version:  # False or None
+                continue
+            elif isinstance(version, str):
+                lib += f"={version}"
+            data_libs = [lib] + data_libs
```

Comment:
This code also seems to be identical to the one in core. Is it necessary here? I'd expect this to be taken care of by the super call

Reply:
Yes, for backwards compatibility. Its not implemented in the base model class of core, I think. This is/ was done in the DataCatalog I think